Upgrading to Swift 4.2

jreyesv opened the pull request:
### What
Upgrading to Swift 4.2 

### How
Running Xcode wizard and replacing use of `hidden` property with `isHidden`

Commit:
Upgrading to Swift 4.2 wizard changes

Commit:
Changing hidden to isHidden

ronnychan-seek commented on stackableTests/HStackTests.swift:
```diff
@@ -52,7 +52,7 @@ class HStackTests: XCTestCase {
         let size2 = CGSize(width: 55, height: 11)
         
-        let stack = HStack(spacing: spacing, layoutMargins: UIEdgeInsetsMake(topMargin, leftMargin, bottomMargin, rightMargin), thingsToStack: [
+        let stack = HStack(spacing: spacing, layoutMargins: UIEdgeInsets.init(top: topMargin, left: leftMargin, bottom: bottomMargin, right: rightMargin), thingsToStack: [
```
is `.init` necessary?

jreyesv replied:
Good point, I will have a look

ronnychan-seek commented on stackableTests/StackTests.swift:
```diff
@@ -69,7 +69,7 @@ class StackTests: XCTestCase {
         let bottomMargin: CGFloat = 10
         
-        let stack = MockStack(layoutMargins: UIEdgeInsetsMake(topMargin, leftMargin, bottomMargin, rightMargin), thingsToStack: [ label1, MockStack(thingsToStack: [ label2 ]), label3 ])
+        let stack = MockStack(layoutMargins: UIEdgeInsets.init(top: topMargin, left: leftMargin, bottom: bottomMargin, right: rightMargin), thingsToStack: [ label1, MockStack(thingsToStack: [ label2 ]), label3 ])
```
same as above

ronnychan-seek commented on stackableTests/UILabel+heightForWidth+Tests.swift:
```diff
@@ -7,7 +7,7 @@ import XCTest
 class UILabel_heightForWidth_Tests: XCTestCase {
     func test_heightForWidth_should_return_expected_result() {
-        let font = UIFont.systemFont(ofSize: 12, weight: UIFontWeightMedium)
+        let font = UIFont.systemFont(ofSize: 12, weight: UIFont.Weight.medium)
```
might be able to use just `.medium`

ronnychan-seek commented on stackableTests/VStackTests.swift:
```diff
@@ -52,7 +52,7 @@ class VStackTests: XCTestCase {
         let height2: CGFloat = 11
         
-        let stack = VStack(spacing: spacing, layoutMargins: UIEdgeInsetsMake(topMargin, leftMargin, bottomMargin, rightMargin), thingsToStack: [
+        let stack = VStack(spacing: spacing, layoutMargins: UIEdgeInsets.init(top: topMargin, left: leftMargin, bottom: bottomMargin, right: rightMargin), thingsToStack: [
```
same as above

ronnychan-seek reviewed (commented):
nice work @jreyesv, some minor questions/suggestions

Commit:
Changes according to PR feedback:
- Removing unnecessary call to init when creating UIEdgeInsets
- Using shorthand for UIFont weight enum

Commit:
Bumping up version and updating home page
